### ISSUE TITLE
swagger-ui-dist "main" property resolves to nothing

### ISSUE THREAD

**Issue by ksmithut** [swagger-api/swagger-ui#3107]
- swagger-ui version: `swagger-ui-dist@3.0.10`
- a swagger file reproducing the issue `N/A`

The `package.json` for the swagger-ui-dist project that gets published to npm has a `main` property of `dist/swagger-ui.js`. In the swagger-ui-dist artifacts, the dist folder does not exist (as the artifacts are likely just the contents of the dist folder). This makes a command like `require.resolve('swagger-ui-dist')` fail.

**Comment by shockey** [comment]
Good catch - this was an artifact from the main project package.json.

IMO best strategy is to set `swagger-ui-bundle.js` as the main file - @ksmithut is that what you were trying to achieve with your require.resolve?

**Comment by ksmithut** [comment]
Essentially, yes. I was setting up an express middleware that would just serve up the built swagger-ui-dist directory, so I was going to so something like this:

```
const path = require('path')
const express = require('express')

const PATH_TO_SWAGGER_UI = path.dirname(require.resolve('swagger-ui-dist'))
// Throws error because it can't resolve './node_modules/swagger-ui-dist/dist/swagger-ui.js'

module.exports = express.static(PATH_TO_SWAGGER_UI)
```

I ended up doing this as a workaround:

```
const PATH_TO_SWAGGER_UI = path.dirname(require.resolve('swagger-ui-dist/index.html'))
```

**Comment by shockey** [comment]
Fixed in #3126!
